### PR TITLE
Fix syntax error from #4164

### PR DESCRIPTION
**Pull request by ptomato**
Function statements require a name.

See #4166

### CODE REVIEW

**Comment by linusg** [comment]
This is still not correct, these all assign the regular `Function` constructor.

File: harness/hidden-constructors.js [hunk 1]
```diff
@@ -15,13 +15,13 @@ var AsyncGeneratorFunction;
 var GeneratorFunction;
 
 try {
-  AsyncFunction = Object.getPrototypeOf(new Function('async function () {}')).constructor;
+  AsyncFunction = Object.getPrototypeOf(new Function('async function dummy() {}')).constructor;
```
Comment (ljharb) [inline harness/hidden-constructors.js:18]:
```suggestion
  AsyncFunction = Object.getPrototypeOf(new Function('return async function dummy() {}')()).constructor;
```

File: harness/hidden-constructors.js [hunk 2]
```diff
@@ -15,13 +15,13 @@ var AsyncGeneratorFunction;
...
 } catch(e) {}
 
 try {
-  AsyncGeneratorFunction = Object.getPrototypeOf(new Function('async function* () {}')).constructor;
+  AsyncGeneratorFunction = Object.getPrototypeOf(new Function('async function* dummy() {}')).constructor;
```
Comment (ljharb) [inline harness/hidden-constructors.js:22]:
```suggestion
  AsyncGeneratorFunction = Object.getPrototypeOf(new Function('return async function* dummy() {}')()).constructor;
```

File: harness/hidden-constructors.js [hunk 3]
```diff
@@ -15,13 +15,13 @@ var AsyncGeneratorFunction;
...
 } catch(e) {}
 
 try {
-  GeneratorFunction = Object.getPrototypeOf(new Function('function* () {}')).constructor;
+  GeneratorFunction = Object.getPrototypeOf(new Function('function* dummy() {}')).constructor;
```
Comment (ljharb) [inline harness/hidden-constructors.js:26]:
```suggestion
  GeneratorFunction = Object.getPrototypeOf(new Function('return function* dummy() {}')()).constructor;
```

**Review by ljharb — APPROVED** [review]
LGTM assuming suggestions are included

**Comment by ptomato** [comment]
Doh! Thanks for the corrections.

**Pull request merged**